check_pgactivity

```diff
@@ -552,24 +549,32 @@ sub find_hostkey($) {
 #  * the ref of the structure to save
 #  * the path to the file storage
 sub save($$$$) {
-    my $host    = shift;
-    my $name    = shift;
-    my $ref     = shift;
-    my $storage = shift;
-    my $all     = {};
-    my $hostkey = find_hostkey($host);
+    my $host     = shift;
+    my $name     = shift;
+    my $ref      = shift;
+    my $storage  = shift;
+    my $all      = {};
+    my $lockfile = "${storage}.lock";
+    my $hostkey  = find_hostkey($host);
 
+    die "no permission to read status file «${storage}»" unless -w ${storage};
```

Comment:
unless I'm missing something it's never going to work if the file isn't already existing?

Also, the error message is wrong.

I think it would be better to do the flock first, and then check for the file existence, create an empty one if needed (or make sure is_storable can handle missing file) or check for correct permissions if it exists.

Reply:
You are right. I believe I fixed this.

I forced push after rebasing the branch, and added a TAP test for hit_ratio that is actually using the status file.